Makefile

```diff
@@ -60,4 +60,15 @@ test:
 	go vet ./...
 	go test ./...
 
+setup-py-env:
+	python3 -m venv pyenv
+
+setup-cli:
+	@[ -f "pyenv/bin/activate" ] || (echo error: python env is not found. Run "make setup-py-env"; exit 1)
+	@[ ! -z "${BASE_URL}" ] || (echo info: BASE_URL is not set. Will use global ARM endpoint)
+	( \
+       source pyenv/bin/activate; \
+	   az extension add --source $$(echo $$(pip install az-cli/src/aro-preview | grep "Stored in directory" | cut -d':' -f2-)/*) -y; \
```

Comment:
I would split it into two targets: one builds a wheel dist (something like `python3 -m venv pyenv && source pyenv/bin/activate && pip install wheel  && python ./setup.py bdist_wheel`) and another just `az extension add --source dist/aro-preview.whl`.

Reply:
Will re-arrange this later :)

Reply:
Asked Elana to take a look at this. Maybe she can give some useful tips on how to make it easier to use.

Reply:
I would suggest doing `pip install -e` in the makefile target above because there is no point in building and rebuilding a wheel when you want your source changes to take effect immediately. Wheels are a distribution format, you should only need to build them when you want to publish the extension.

Reply:
@ehashman `az extension add` only accepts `.whl`s [is seems](https://github.com/Azure/azure-cli/blob/3fa3de4fe7ddff0f58a33eb6ae62e358c91718e6/src/azure-cli-core/azure/cli/core/extension/operations.py#L83-L84).

If we want changes to take effect immidiately we can look into building the extension into [the `EXTENSIONS_DIR` dir](https://github.com/Azure/azure-cli/blob/3fa3de4fe7ddff0f58a33eb6ae62e358c91718e6/src/azure-cli-core/azure/cli/core/extension/__init__.py#L19) and skip calling `az extension add`. But I don't think it's important: as far as I understand, client is not going to change often: only when API changes. Or am I missing something?

Reply:
There is a little bit more to that. I still need to find out how we can do this without rebuilding this constantly. Like:
1. Change CLI
2. Test
3. goTo1

This would not work as it is now.